migration/20170522-2-move-configuration-to-external-integrations.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+"""Move integration details from the Configuration file into the
+database as ExternalIntegrations
+"""
+import os
+import sys
+import logging
+from nose.tools import set_trace
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from config import Configuration
+from model import (
+    ExternalIntegration,
+    get_one_or_create,
+    production_session,
+)
+
+log = logging.getLogger(name="Core configuration import")
+
+def log_import(service_name):
+    log.info("Importing configuration for %s" % service_name)
+
+try:
+    Configuration.load()
+    _db = production_session()
+
+    # Import CDN configuration.
+    cdn_conf = Configuration.integration(Configuration.CDN_INTEGRATION)
+    if cdn_conf:
+        service, ignore = get_one_or_create(
+            _db, ExternalIntegration, provider=ExternalIntegration.CDN
+        )
+        [service.set_setting(k, v) for k, v in cdn_conf.items()]
+
+    # Import Circulation Manager configuration.
+    circulation_manager_conf = Configuration.integration(
+        Configuration.CIRCULATION_MANAGER_INTEGRATION
+    )
+    if circulation_manager_conf:
+        url = circulation_manager_conf.get(Configuration.URL)
+
+        log_import(ExternalIntegration.CIRCULATION_MANAGER)
+        service, ignore = get_one_or_create(
+            _db, ExternalIntegration, provider=ExternalIntegration.CIRCULATION_MANAGER
+        )
+        service.url = unicode(url)
+
+    # Import Content Server configuration.
+    content_server_conf = Configuration.integration(
+        Configuration.CONTENT_SERVER_INTEGRATION
+    )
+
+    if content_server_conf:
+        url = content_server_conf.get(Configuration.URL)
+
+        log_import(ExternalIntegration.CONTENT_SERVER)
+        service, ignore = get_one_or_create(
+            _db, ExternalIntegration, provider=ExternalIntegration.CONTENT_SERVER
+        )
+        service.url = unicode(url)
```

Comment:
Is there a "content server integration" that's not a Collection with protocol=OPDS_IMPORT? Maybe there is for metadata.